### PR TITLE
InProcessSolidResult.output_value

### PR DESCRIPTION
**Pull request by sryza**
## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution/execution_results.py [hunk 1]
```diff
@@ -103,9 +104,35 @@ def output_values(self) -> Dict[str, Any]:
...
+                in the normal case, and a dict of mapping keys to values in the mapped case.
+        """
+        check.str_param(output_name, "output_name")
+
+        if not self._output_capture:
+            raise DagsterInvariantViolationError(
+                "Trying to fetch output values, but output_capturing_enabled was False for the run"
+            )
+
+        return self._output_capture[StepOutputHandle(str(self.handle), output_name)]
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/execution/execution_results.py:134]: I don't think this will work for the dynamic case

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Add a test for the dynamic out case

File: python_modules/dagster/dagster/core/execution/execution_results.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster/core/execution/execution_results.py]: Good call - just updated.

**Review by dpeng817 — APPROVED** [review]
Nice. Probably gonna need to rebase on top of https://github.com/dagster-io/dagster/pull/4646 but LGTM.

**Pull request merged**